Broken Fedora Build

https://github.com/baresip/baresip/actions/runs/11626174484/job/32377361314?pr=3182

@robert-scheck can you have a look?

```
+ /usr/bin/cmake -S . -B redhat-linux-build -DCMAKE_C_FLAGS_RELEASE:STRING=-DNDEBUG -DCMAKE_CXX_FLAGS_RELEASE:STRING=-DNDEBUG -DCMAKE_Fortran_FLAGS_RELEASE:STRING=-DNDEBUG -DCMAKE_VERBOSE_MAKEFILE:BOOL=ON -DCMAKE_INSTALL_DO_STRIP:BOOL=OFF -DCMAKE_INSTALL_PREFIX:PATH=/usr -DINCLUDE_INSTALL_DIR:PATH=/usr/include -DLIB_INSTALL_DIR:PATH=/usr/lib64 -DSYSCONF_INSTALL_DIR:PATH=/etc -DSHARE_INSTALL_PREFIX:PATH=/usr/share -DLIB_SUFFIX=64 -DBUILD_SHARED_LIBS:BOOL=ON
CMake Error: The source directory "/__w/baresip/baresip/re/placeholder-0-build" does not appear to contain CMakeLists.txt.
Specify --help for usage, or press the help button on the CMake GUI.
error: Bad exit status from /var/tmp/rpm-tmp.rUN1wO (%build)
    Bad exit status from /var/tmp/rpm-tmp.rUN1wO (%build)

RPM build errors:
Error: Process completed with exit code 1.
```

--- a/.github/workflows/fedora.yml
+++ b/.github/workflows/fedora.yml
@@ -35,10 +35,12 @@ jobs:
   build:
     needs: dependency
     runs-on: ubuntu-latest
-    container: fedora:40
+    container: fedora
 
     steps:
     - uses: actions/checkout@v4
+      with:
+        path: placeholder-0-build
 
     - name: "Install build-time requirements"
       run: |
@@ -94,9 +96,9 @@ jobs:
     - uses: actions/download-artifact@v4
       with:
         name: re-fedora
-        path: re
+        path: re/placeholder-0-build
 
-    - name: "Write placeholer RPM spec file"
+    - name: "Write placeholder RPM spec file"
       # Use same placeholder RPM spec file until re and baresip need different %build commands
       run: |
         cat > placeholder.spec <<EOF
@@ -120,9 +122,7 @@ jobs:
         EOF
 
     - name: "Write RPM macros configuration for re"
-      run: |
-        echo "%_builddir ${PWD}/re" > "${HOME}/.rpmmacros"
-        echo "%buildroot ${PWD}/re/%{_vendor}-%{_target_os}-install" >> "${HOME}/.rpmmacros"
+      run: echo "%_builddir ${PWD}/re" > "${HOME}/.rpmmacros"
 
     - name: "RPM %build re"
       run: rpmbuild -bc --short-circuit placeholder.spec
@@ -133,16 +133,23 @@ jobs:
 
     - name: "Relocate re from RPM %{buildroot} to /"
       run: |
-        cp -avT $(rpm --eval '%{buildroot}/') /  # Work around '%{buildroot} can not be "/"' from RPM
+        cp -avT "${PWD}/re/placeholder-0-build/BUILDROOT/" /  # Intermediate builddir since RPM 4.20
+        # isn't available externally, see https://github.com/rpm-software-management/rpm/issues/3147
         rm -rf re  # Baresip must not pick up this directory (or wrong rpaths are written to binaries)
 
     - name: "Run ldconfig"
       run: ldconfig
 
-    - name: "Write RPM macros configuration for baresip"
+    - name: "Add selftest to placeholder RPM spec file"
       run: |
-        echo "%_builddir ${PWD}" > "${HOME}/.rpmmacros"
-        echo "%buildroot ${PWD}/%{_vendor}-%{_target_os}-install" >> "${HOME}/.rpmmacros"
+        cat >> placeholder.spec <<EOF
+        %check
+        cd %{__cmake_builddir}
+        ./test/selftest -d ../test/data/ -v
+        EOF
+
+    - name: "Write RPM macros configuration for baresip"
+      run: echo "%_builddir ${PWD}" > "${HOME}/.rpmmacros"
 
     - name: "RPM %build baresip"
       run: rpmbuild -bc --short-circuit placeholder.spec

